pysrc/juliacall/__init__.py

```diff
@@ -248,8 +248,37 @@ def jlstr(x):
                 "PYTHON_JULIACALL_HANDLE_SIGNALS=no."
             )
 
-init()
+    # Next, automatically load the juliacall extension if we are in a Jupyter notebook
+    CONFIG['autoload_extensions'] = choice('autoload_extensions', ['yes', 'no'])[0]
+    if CONFIG['autoload_extensions'] in {'yes', None}:
+        try:
+            get_ipython = sys.modules['IPython'].get_ipython
+
+            if 'IPKernelApp' not in get_ipython().config:
+                raise ImportError('console')
+
+            if CONFIG['autoload_extensions'] is None:
+                # Only let the user know if it was not explicitly set
+                print(
+                    "Detected Jupyter notebook. Loading juliacall extension. To disable, you can either "
```

Comment:
The IPython kernel can be used in places other than Jupyter, so can you make the first bit more general please? Perhaps "IPython detected, loading juliacall extension."

Reply:
The current autoloading only applies to Jupyter notebooks. Want me to make it work for IPython as well?

Reply:
Edit: just added it for IPython as well. My only fear in making it more general is that there might be other types of IPython notebooks that break... But I guess we will see.